06-building-decays.md

```diff
@@ -149,8 +186,26 @@ from Configurables import DaVinci
 DaVinci().UserAlgorithms += [dstar_seq.sequence()]
 ```
 
+
+> ## Debugging your selection chain {.callout}
+> The `PhysSelPython.Wrappers` offers a very useful utility for debugging your selection chains, called `PrintSelection`.
+> It gets a `Selection` as input and it can be used the same way, except it will print the decay tree everytime making use of the `PrintDecayTree` algorithm which was discussed in the [Exploring a DST](05-interactive-dst.html) lesson.
+>
+> For more complex debugging, one can setup `DaVinci` with `graphviz
+> ```shell
+> SetupDaVinci --use "graphviz v* LCG_Interfaces"
+> ```
+>
+> And add a nice graphical representation of the selection sequence
```

Comment:
I was thinking of leaving that as an exercise for them to find out... What
do you think?

On Wed, Jun 3, 2015 at 10:14 AM, Tim Head <notifications@github.com> wrote:

> In 06-building-decays.md
> <https://github.com/lhcb/first-analysis-steps/pull/90#discussion_r31601888>
> :
>
> > @@ -149,8 +186,26 @@ from Configurables import DaVinci
> >  DaVinci().UserAlgorithms += [dstar_seq.sequence()]
> >  ```
> >
> > +
> > +> ## Debugging your selection chain {.callout}
> > +> The `PhysSelPython.Wrappers` offers a very useful utility for debugging your selection chains, called `PrintSelection`.
> > +> It gets a `Selection` as input and it can be used the same way, except it will print the decay tree everytime making use of the `PrintDecayTree` algorithm which was discussed in the [Exploring a DST](05-interactive-dst.html) lesson.
> > +>
> > +> For more complex debugging, one can setup `DaVinci` with `graphviz
> > +> ```shell
> > +> SetupDaVinci --use "graphviz v* LCG_Interfaces"
> > +> ```
> > +>
> > +> And add a nice graphical representation of the selection sequence
>
> Add a few sentences on what this actually does. Creates a PNG of the graph
> of algorithms, etc. It is obvious if you know graphviz etc but if you don't
> it is less obvious.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lhcb/first-analysis-steps/pull/90/files#r31601888>.
>



-- 
Dr. Albert Puig Navarro
Laboratoire de Physique des Hautes Energies
Ecole Polytechnique Fédérale de Lausanne (EPFL)
BSP 614.4 (Cubotron UNIL) CH-1015 Lausanne
EPFL Phone: 021 6939808
CERN Phone: 72518